jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamSchemasResultSet.java

```diff
@@ -99,7 +99,8 @@ private void populateCurrentRows(TimestreamConnection connection, String schemaP
       final String query = "SHOW DATABASES" +
               (Strings.isNullOrEmpty(schemaPattern) ? "" : " LIKE '" + schemaPattern + "'");
       try (ResultSet rs = statement.executeQuery(query)) {
-        while (rs != null && rs.next()) {
+        // ResultSet is not known to be null, so no null check is needed here
+        while (rs.next()) {
```

Comment:
ResultSet is not known to be null, so no null check is needed here. Spotbugs would give extra errors with the null check

Reply:
Curious Spotbugs gives an error for this null check. If it is not needed, it should not be an error.

Reply:
Spotbug is very picky about things, and it complained about the null check because ResultSets cannot actually be null. The only case when `rs` is null is when the Mockito does not return anything in the mock tests.